src/openfermion/measurements/_rdm_mapping_functions_test.py

```diff
@@ -0,0 +1,187 @@
+                test_oqdm = opdm_to_oqdm(molecule.fci_one_rdm)
+                true_oqdm = numpy.eye(molecule.n_qubits) - molecule.fci_one_rdm
+                assert numpy.allclose(test_oqdm, true_oqdm)
+
+    def test_oqdm_to_opdm(self):
+        for file in filter(lambda x: x.endswith(".hdf5"),
+                           os.listdir(DATA_DIRECTORY)):
+            molecule = MolecularData(
+                filename=os.path.join(DATA_DIRECTORY, file))
+            if molecule.fci_one_rdm is not None:
+                true_oqdm = numpy.eye(molecule.n_qubits) - molecule.fci_one_rdm
+                test_opdm = oqdm_to_opdm(true_oqdm)
+                assert numpy.allclose(test_opdm, molecule.fci_one_rdm)
+
+    def test_tqdm_conversions(self):
+        # construct the 2-hole-RDM for LiH the slow way
+        # TODO: speed up this calculation by directly contracting from the wf.
+        file = "H2_6-31g_singlet_0.75.hdf5"
+        molecule = MolecularData(filename=os.path.join(DATA_DIRECTORY, file))
+        true_tqdm = self.tqdm_h2_6_31g
+
+        test_tqdm = tpdm_to_tqdm(molecule.fci_two_rdm, molecule.fci_one_rdm)
+        assert numpy.allclose(true_tqdm, test_tqdm)
+
+        true_oqdm = numpy.eye(molecule.n_qubits) - molecule.fci_one_rdm
+        test_oqdm = tqdm_to_oqdm(true_tqdm,
+                                 molecule.n_qubits - molecule.n_electrons)
+        assert numpy.allclose(true_oqdm, test_oqdm)
+
+        test_tpdm = tqdm_to_tpdm(true_tqdm, molecule.fci_one_rdm)
+        assert numpy.allclose(test_tpdm, molecule.fci_two_rdm)
+
+        file = "H2_sto-3g_singlet_1.4.hdf5"
+        molecule = MolecularData(filename=os.path.join(DATA_DIRECTORY, file))
+        true_tqdm = self.tqdm_h2_sto3g
+
+        test_tqdm = tpdm_to_tqdm(molecule.fci_two_rdm, molecule.fci_one_rdm)
+        assert numpy.allclose(true_tqdm, test_tqdm)
+
+        true_oqdm = numpy.eye(molecule.n_qubits) - molecule.fci_one_rdm
+        test_oqdm = tqdm_to_oqdm(true_tqdm,
+                                 molecule.n_qubits - molecule.n_electrons)
+        assert numpy.allclose(true_oqdm, test_oqdm)
+
+        test_tpdm = tqdm_to_tpdm(true_tqdm, molecule.fci_one_rdm)
+        assert numpy.allclose(test_tpdm, molecule.fci_two_rdm)
+
+        file = "H1-Li1_sto-3g_singlet_1.45.hdf5"
+        molecule = MolecularData(filename=os.path.join(DATA_DIRECTORY, file))
+        true_tqdm = self.tqdm_lih_sto3g
+
+        test_tqdm = tpdm_to_tqdm(molecule.fci_two_rdm, molecule.fci_one_rdm)
+        assert numpy.allclose(true_tqdm, test_tqdm)
+
+        true_oqdm = numpy.eye(molecule.n_qubits) - molecule.fci_one_rdm
+        test_oqdm = tqdm_to_oqdm(true_tqdm,
+                                 molecule.n_qubits - molecule.n_electrons)
+        assert numpy.allclose(true_oqdm, test_oqdm)
+
+        test_tpdm = tqdm_to_tpdm(true_tqdm, molecule.fci_one_rdm)
+        assert numpy.allclose(test_tpdm, molecule.fci_two_rdm)
+
+    def test_phdm_conversions(self):
```

Comment:
This test also has 9 asserts and redefines the builtin `file`.

Reply:
Indeed, it is much easier to identify the problem when these are separate tests.